vars/buildAndAssemble.groovy

```diff
@@ -0,0 +1,40 @@
+void call(Map args = [:]) {
+    String manifest = args.manifest ?: "manifests/${INPUT_MANIFEST}"
+
+    buildManifest(
+        args + [
+            manifest: manifest,
+            lock: true
+        ]
+    )
+
+    String manifestLock = args.dryRun ? manifest : "${manifest}.lock"
+    manifestSHA = sha1(manifestLock)
+    echo "Manifest SHA: ${manifestSHA}"
+
+    lib = library(identifier: 'jenkins@20211122', retriever: legacySCM(scm))
+    def inputManifest = lib.jenkins.InputManifest.new(readYaml(file: manifestLock))
+    String shasRoot = inputManifest.getSHAsRoot("${JOB_NAME}", args.platform, args.architecture)
+
+    buildManifest(
+        args + [
```

Comment:
Can use `args.extend([manifest: manifestLock])`

Reply:
Isn't that in-place? Either way it doesn't seem any better, or is it?

Reply:
Actually `+` creates a new list and `extends` add it in the existing list. There's a good article for the comparison https://blog.finxter.com/python-list-concatenation-add-vs-inplace-add-vs-extend/

Reply:
Right, I do not want to modify the value being passed in.